kanban/api/views/issue.py

```diff
@@ -0,0 +1,67 @@
+from kanban.api.serializers import IssueSerializer
+from kanban.models.issue import Issue
+from kanban.tasks import (
+    issue_assignee_change_notification,
+    issue_due_time_notification,
+)
+from rest_framework import viewsets, generics
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from user_account.api.serializers import AssignUserSerializer
+from user_account.models.user_account import UserAccount
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+
+
+class IssueViewSet(viewsets.ModelViewSet):
+    queryset = Issue.objects.select_related("assignee", "project")
+    serializer_class = IssueSerializer
+
+    def perform_update(self, serializer):
+        previous_due_date = serializer.instance.due_date
+        instance = serializer.save()
+
+        if instance.due_date != previous_due_date and instance.assignee is not None:
+            issue_due_time_notification.apply_async(
+                [instance.pk], eta=instance.due_date
+            )
+
+        return instance
+
+
+class IssueAssignee(generics.ListCreateAPIView):
+    queryset = UserAccount.objects.all()
+    serializer_class = AssignUserSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(issues__pk=self.kwargs["pk"])
+
+    def post(self, request, *args, **kwargs):
+        issue = get_object_or_404(Issue, id=self.kwargs.get("pk"))
+        user = get_object_or_404(UserAccount, email=request.data.get("email"))
+
+        if not issue.project.users.filter(pk=user.pk).exists():
+            return Response(
+                {"response": "User must belong to the project"},
+                status=status.HTTP_406_NOT_ACCEPTABLE,
```

Comment:
406 should not be used for validation, it refers to `Accept` header of the request only